woocommerce/woocommerce
# Merge Shipping Settings Legacy Refresh

### Changes proposed in this Pull Request:

This PR merges work to refresh legacy Shipping Settings. See more information on the project in pevPIe-hv-p2

### How to test the changes in this Pull Request:

This feature branch has had all its individual parts tested by the following PRs.

The updated flow is the entirety of Shipping Settings, ie WooCommerce > Settings > Shipping.

* https://github.com/woocommerce/woocommerce/pull/40184
* https://github.com/woocommerce/woocommerce/pull/40265
* https://github.com/woocommerce/woocommerce/pull/40421
* https://github.com/woocommerce/woocommerce/pull/40554
* https://github.com/woocommerce/woocommerce/pull/40615
* https://github.com/woocommerce/woocommerce/pull/40724
* https://github.com/woocommerce/woocommerce/pull/40816
* https://github.com/woocommerce/woocommerce/pull/40838
* https://github.com/woocommerce/woocommerce/pull/40860
* https://github.com/woocommerce/woocommerce/pull/40893
* https://github.com/woocommerce/woocommerce/pull/40982
* https://github.com/woocommerce/woocommerce/pull/41015
* https://github.com/woocommerce/woocommerce/pull/41035
* https://github.com/woocommerce/woocommerce/pull/41054
* https://github.com/woocommerce/woocommerce/pull/41445
* https://github.com/woocommerce/woocommerce/pull/41446
* https://github.com/woocommerce/woocommerce/pull/41481
* https://github.com/woocommerce/woocommerce/pull/41529

### Flows to test

When checking flows, read all copy, play with the UI inputs, buttons, etc. Be sure to add, then edit methods, zones, and classes to ensure data persistence. Do table styles look ok? Does the drag n drop work? Does data persist there? Do the modals for data entry work properly?

1. Add a shipping zone `/wp-admin/admin.php?page=wc-settings&tab=shipping&zone_id=new`. Create a new zone, pick a region and add a Local pickup, fat rate, and free shipping methods.
2. Add a shipping class `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=classes`. Be sure to return to your shipping zone's flat rate method - new fields should be added relating to shipping classes.
3. Check extensibility of shipping methods. Install and activate [Flat Rate Boxes](https://github.com/woocommerce/woocommerce-shipping-flat-rate-boxes/releases) to see an additional method added.
4. More extensibility. Install and activate [Fedex Shipping](https://github.com/woocommerce/woocommerce-shipping-fedex) and add/edit shipping classes. You'll need a US address and USD currency for this plugin. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=fedex` and click Enable freight then save.
5. Region picker - Use the region picker at `/wp-admin/admin.php?page=wc-settings&tab=shipping&zone_id=new`. Can you search for all values? How about searching for values with accents? Does the keyboard interaction work for navigating the tree of countries? How about selecting/unselecting values?

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Update Shipping Settings to new experience.

#### Comment 

</details>
